tests/unit/HeadlessLayerImplicitAnimationTests.swift

```diff
@@ -83,67 +132,125 @@ class HeadlessLayerImplicitAnimationTests: XCTestCase {
   }
 
   func testCATransactionTimingTakesPrecedenceOverUIViewTimingInside() {
+    let layer = CALayer()
+    window.layer.addSublayer(layer)
+    CATransaction.flush()
+
     UIView.animate(withDuration: 0.5) {
-      CATransaction.begin()
-      CATransaction.setAnimationDuration(0.2)
-      self.layer.opacity = 0.5
-      CATransaction.commit()
+      layer.opacity = 0.5
     }
 
-    let animation = layer.animation(forKey: "opacity") as! CABasicAnimation
-    XCTAssertEqual(animation.keyPath, "opacity")
-    XCTAssertEqual(animation.duration, 0.2)
+    XCTAssert(layer.animation(forKey: "opacity") is CABasicAnimation)
+    if let animation = layer.animation(forKey: "opacity") as? CABasicAnimation {
+      XCTAssertEqual(animation.keyPath, "opacity")
+      XCTAssertEqual(animation.duration, CATransaction.animationDuration())
+    }
   }
 
   // Verifies the somewhat counter-intuitive fact that CATransaction's animation duration always
   // takes precedence over UIView's animation duration. This means that animating a headless layer
   // using UIView animation APIs may not result in the expected traits.
   func testCATransactionTimingTakesPrecedenceOverUIViewTimingOutside() {
-    CATransaction.begin()
-    CATransaction.setAnimationDuration(0.2)
-    UIView.animate(withDuration: 0.5) {
-      self.layer.opacity = 0.5
+    let layer = CALayer()
+    window.layer.addSublayer(layer)
+    CATransaction.flush()
+
+    UIView.animate(withDuration: CATransaction.animationDuration() + 0.1) {
+      layer.opacity = 0.5
     }
-    CATransaction.commit()
 
-    let animation = layer.animation(forKey: "opacity") as! CABasicAnimation
-    XCTAssertEqual(animation.keyPath, "opacity")
-    XCTAssertEqual(animation.duration, 0.2)
+    XCTAssert(layer.animation(forKey: "opacity") is CABasicAnimation)
+    if let animation = layer.animation(forKey: "opacity") as? CABasicAnimation {
+      XCTAssertEqual(animation.keyPath, "opacity")
+      XCTAssertEqual(animation.duration, CATransaction.animationDuration())
+    }
   }
 
   func testDoesImplicitlyAnimateInUIViewAnimateBlock() {
```

Comment:
Maybe comments will help to distinguish these cases? Looks just like `testImplicitlyAnimatesInUIViewAnimateBlockWithCATransactionDuration` to me.